CNTRLPLANE-194: add auth-api-bootstrap stage to bootkube.sh template

everettraven opened the pull request:
## Description
Adds a new bootstrapping stage for the cluster-authentication-operator to render the `RoleBindingRestriction` CRD to the set of bootstrap manifests.

## Motivation
We are working on moving the management of the `RoleBindingRestriction` CRD to the cluster-authentication-operator instead of it being managed by CVO as outlined in https://github.com/openshift/enhancements/pull/1726. The `RoleBindingRestriction` CRD is required in the bootstrapping manifests as the `authorization.openshift.io/RestrictSubjectBindings` admission plugin will reject creation of `system:*` RoleBinding resources during the installation process if the `RoleBindingRestriction` API does not exist.

I did some testing to ensure that we can gradually transition ownership without clashing with CVO by merging, in order:
- https://github.com/openshift/cluster-authentication-operator/pull/748
- This PR
- https://github.com/openshift/api/pull/2138

everettraven commented:
Can't merge until https://github.com/openshift/cluster-authentication-operator/pull/748 has merged first.

/hold

liouk commented on data/data/bootstrap/files/usr/local/bin/bootkube.sh.template:
```diff
@@ -117,6 +118,37 @@ then
+		--volume "$PWD:/assets:z" \
+		"${AUTH_OPERATOR_IMAGE}" \
+		render \
+		--asset-output-dir=/assets/auth-api-bootstrap/manifests \
```
End command:
```suggestion
		--asset-output-dir=/assets/auth-api-bootstrap/manifests
```

everettraven replied:
Good catch! Updated in the latest push.

liouk commented:
/lgtm

patrickdillon commented:
/approve

If you could update your commit message according to our guidelines that would be great: https://github.com/openshift/installer/blob/main/CONTRIBUTING.md#commit-message-format

The text of your PR description should work well.

everettraven commented:
@patrickdillon I updated the commit message and added some flags that we recently added in https://github.com/openshift/cluster-authentication-operator/pull/748/files#diff-9169c482883c2b2e5f4038e6cd3c2505765da290271b6c1c01c5dc847266e59bR31-R38

This PR should still be held until the cluster-authentication-operator PR merges, but feel free to let me know if anything else needs any further updates.

Thanks!

liouk commented:
/lgtm

everettraven commented:
/hold cancel

/retest

vrutkovs commented on data/data/bootstrap/files/usr/local/bin/bootkube.sh.template:
```diff
@@ -117,6 +118,41 @@ then
+		render \
+		--asset-output-dir=/assets/auth-api-bootstrap/manifests \
+		--rendered-manifest-dir=/assets/manifests \
+		--cluster-profile=${CLUSTER_PROFILE_ANNOTATION} \
```
This seems to breaks nightlies:
```
Feb 24 08:13:18 ip-10-0-103-202 bootkube.sh[124202]: Rendering auth api manifests...
Feb 24 08:13:18 ip-10-0-103-202 bootkube.sh[124202]: /usr/local/bin/bootkube.sh: line 134: CLUSTER_PROFILE_ANNOTATION: unbound variable
Feb 24 08:13:18 ip-10-0-103-202 systemd[1]: bootkube.service: Main process exited, code=exited, status=1/FAILURE

```
